data/src/main/java/me/androidbox/data/local/event/EventRepositoryImp.kt

```diff
@@ -0,0 +1,64 @@
+package me.androidbox.data.local.event
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.onEach
+import me.androidbox.data.local.dao.EventDao
+import me.androidbox.data.local.entity.EventEntity
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.domain.authentication.model.Event
+import me.androidbox.domain.authentication.remote.EventRepository
+import javax.inject.Inject
+
+class EventRepositoryImp @Inject constructor(
+    private val eventDao: EventDao
+) : EventRepository {
+
+    override fun getEventsFromTimeStamp(
+        startTimeStamp: Long,
+        endTimeStamp: Long
+    ): Flow<ResponseState<List<Event>>> {
+        return flow<ResponseState<List<Event>>> {
+            eventDao.getEventsFromTimeStamp(startTimeStamp, endTimeStamp).onEach { listOfEventEntity ->
+                val listOfEvent = listOfEventEntity.map { eventEntity ->
+                    Event(
+                        id = eventEntity.id,
+                        title = eventEntity.title,
+                        description = eventEntity.description,
+                        from = eventEntity.from,
+                        to = eventEntity.to,
+                        remindAt = eventEntity.remindAt,
+                        host = eventEntity.host,
+                        isUserEventCreator = eventEntity.isUserEventCreator,
+                        attendees = eventEntity.attendees,
+                        photos = eventEntity.photos
+                    )
+                }
```

Comment:
Here, I'd recommend to create a mapper. In this app you'll have this logic quite often (to convert an entity to a domain model) and with mappers you make it reusable:
```
// In EventMappers.kt

fun EventEntity.toEvent(): Event {
    return Event(...)
}

// EventRepositoryImp.kt
val listOfEvents = listOfEventEntities.map { it.toEvent() }
```